### PR TITLE
20231011 cbmc 5.94.0

### PR DESCRIPTION
**Pull request by TGWDB**
PR for `cbmc` 5.94.0. Note that this is trialing including a `CHANGELOG` as discussed in #7907 .

The following have committed PRs and should review my editing of their PR title in the `CHANGELOG` to ensure I didn't mangle it: @thomasspriggs @peterschrammel @tautschnig @esteffin @NlightNFotis @qinheping @kroening 

- [ ] Each commit message has a non-empty body, explaining why the change was made.
- [ ] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [ ] The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [ ] My commit message includes data points confirming performance improvements (if claimed).
- [ ] My PR is restricted to a single feature or bugfix.
- [ ] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

File: CHANGELOG [hunk 1]
```diff
@@ -0,0 +1,22 @@
+## What's Changed
+* Increase support for unions in incremental smt2 decision procedure by @thomasspriggs in https://github.com/diffblue/cbmc/pull/7926
```
Comment (thomasspriggs) [inline CHANGELOG]: :pick: I don't think we have added enough support for any union examples to work from a user perspective.

File: CHANGELOG [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+## Bug Fixes
+* Fix problem on array size L2 renaming by @esteffin in https://github.com/diffblue/cbmc/pull/7877
+* Fix shadow memory missing aggregation on non-compound bitvector types by @esteffin in https://github.com/diffblue/cbmc/pull/7935
+
+## Internal Improvements
+* Interval expressions: cleanup type hierarchy by @tautschnig in https://github.com/diffblue/cbmc/pull/7925
+* Rename {min,max}_exprt to {min,max}_value_exprt by @tautschnig in https://github.com/diffblue/cbmc/pull/7931
+* Introduce widen_if_needed by @kroening in https://github.com/diffblue/cbmc/pull/7938
+* format_expr: format constants with natural number type by @kroening in https://github.com/diffblue/cbmc/pull/7940
+* Enable all struct tests under `regression/cbmc/` for new SMT backend by @NlightNFotis in https://github.com/diffblue/cbmc/pull/7809
```
Comment (thomasspriggs) [inline CHANGELOG]: :pick: I suspect this is actually a misleadingly named PR, as we enabled the tests which pass only. There are at least some remaining to fix.

**Review by thomasspriggs — APPROVED** [review]

File: CHANGELOG [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+* Shadow memory addresses now also support --pointer-check argument by @esteffin in https://github.com/diffblue/cbmc/pull/7936
+* [DOCS] Add documentation on CBMC Shadow Memory intrinsics by @NlightNFotis in https://github.com/diffblue/cbmc/pull/7913
+* [DOCS] Improve doxygen for shadow_memory_util by @NlightNFotis in https://github.com/diffblue/cbmc/pull/7930
+* [SYNTHESIZER] Allow goto-synthesizer accept all CBMC options  by @qinheping in https://github.com/diffblue/cbmc/pull/7900
+
+## Bug Fixes
+* Fix problem on array size L2 renaming by @esteffin in https://github.com/diffblue/cbmc/pull/7877
+* Fix shadow memory missing aggregation on non-compound bitvector types by @esteffin in https://github.com/diffblue/cbmc/pull/7935
+
+## Internal Improvements
```
Comment (tautschnig) [inline CHANGELOG]: I suggest not even including this section, though perhaps those could be listed in the PR description for reviewers to potentially override whoever is putting together the PR. My personal take is that the changelog should strongly focus on observable changes, where "observable" are all those changes that impact users in the first instance and developers if there are substantial API changes.

File: CHANGELOG [hunk 4]
```diff
@@ -0,0 +1,22 @@
+## What's Changed
+* Increase support for unions in incremental smt2 decision procedure by @thomasspriggs in https://github.com/diffblue/cbmc/pull/7926
+* Add ASV book CBMC examples by @peterschrammel in https://github.com/diffblue/cbmc/pull/7921
```
Comment (tautschnig) [inline CHANGELOG]: Not relevant to users.

File: CHANGELOG [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+* Add models for C library: getpwnam, getpwuid, and getopt by @tautschnig in https://github.com/diffblue/cbmc/pull/7919 and https://github.com/diffblue/cbmc/pull/7916
+* Shadow memory addresses now also support --pointer-check argument by @esteffin in https://github.com/diffblue/cbmc/pull/7936
+* [DOCS] Add documentation on CBMC Shadow Memory intrinsics by @NlightNFotis in https://github.com/diffblue/cbmc/pull/7913
+* [DOCS] Improve doxygen for shadow_memory_util by @NlightNFotis in https://github.com/diffblue/cbmc/pull/7930
```
Comment (tautschnig) [inline CHANGELOG]: I think those two items could be combined.

File: CHANGELOG [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+* [SYNTHESIZER] Allow goto-synthesizer accept all CBMC options  by @qinheping in https://github.com/diffblue/cbmc/pull/7900
```
Comment (tautschnig) [inline CHANGELOG]: I'd reword this as "goto-synthesizer now accepts all CBMC options by @qinheping ..."

File: CHANGELOG [hunk 7]
```diff
@@ -0,0 +1,22 @@
+## What's Changed
```
Comment (tautschnig) [inline CHANGELOG:3]: Before that line we'll need a `#`-level heading that states the version that these changes are referring to.

**Review by tautschnig — APPROVED** [review]
Thank you for introducing a changelog!

**Review by esteffin — APPROVED** [review]

**Review by NlightNFotis — APPROVED** [review]

**Comment by tautschnig** [comment]
Could we please get #7954 into this release (i.e., hold off releasing until this is approved, and then rebase this release)?
